### PR TITLE
Remove getset

### PR DESCRIPTION
**Pull request by serprex**
getset & its proc-macro-error dependencies seem unmaintained (https://github.com/jbaublitz/getset/pull/84#issuecomment-1087770476 https://github.com/jbaublitz/getset/pull/87#issuecomment-1167658416)

In particular, I'm not confident I can get upstream to update to syn 2

I also don't think getter/setter patterns make particular sense in Rust

Use `#[non_exhausitive]` in places to maintain capability to add fields without breaking semver

In some places I omitted `#[non_exhaustive]` where I didn't think it should be necessary, if you disagree let me know & I'll apply it more widely. Similarly, just let me know if you don't think `#[non_exhaustive]` is necessary

In some cases only getters were implemented, in those cases I implemented getter methods by hand & left the fields non-public

### CODE REVIEW

File: src/messages/copy.rs [hunk 1]
```diff
@@ -60,8 +58,7 @@ impl Message for CopyDone {
 
 pub const MESSAGE_TYPE_BYTE_COPY_FAIL: u8 = b'f';
 
-#[derive(Getters, Setters, MutGetters, PartialEq, Eq, Debug, Default, new)]
-#[getset(get = "pub", set = "pub", get_mut = "pub")]
+#[derive(PartialEq, Eq, Debug, Default, new)]
 pub struct CopyFail {
     message: String,
```
Comment (sunng87) [inline src/messages/copy.rs:63]: Do we need to make members of all message structs `pub` for consistency?

**Comment by sunng87** [comment]
@serprex Thank you! I have checked all usage of `#[non_exhausitive]` I think it's just right for all necessary structs. 

While I think it's ok to have direct member access over getter/setter, in particular when `getset` development become stalled, I need to think about impact of this because it's a breaking change.

File: src/messages/copy.rs (hunk 1 above)
Comment (serprex) [reply src/messages/copy.rs:63]: Yes, I missed this one

**Comment by serprex** [comment]
If you don't want to have a breaking change I can look to create a minimal getset macro based on the rather straightforward use of the crate here

**Review by sunng87 — APPROVED** [review]

**Comment by sunng87** [comment]
I just realized that a major drawback of pub struct + pub field access is it stops from adding new fields in semver non-breaking change. Because the initializer syntax `Struct { field: value ... }` is published as API.

**Comment by serprex** [comment]
That is prevented by
```rust
#[non_exhaustive]
```

**Comment by sunng87** [comment]
Yes, we need to add this attribute carefully on each struct during 0.x phase. Adding this attribute is a breaking change too.

**Pull request merged**